libsignal-service-actix/src/push_service.rs

```diff
@@ -107,6 +107,15 @@ impl AwcPushService {
                 })?;
                 Err(ServiceError::StaleDevices(stale_devices))
             },
+            StatusCode::PRECONDITION_REQUIRED => {
+                let proof_required = response.json().await.map_err(|e| {
+                    log::error!("Failed to decode HTTP 428 response: {}", e);
+                    ServiceError::UnhandledResponseCode {
+                        http_code: StatusCode::PRECONDITION_REQUIRED.as_u16(),
+                    }
+                })?;
+                Err(ServiceError::ProofRequiredError(proof_required))
+            },
```

Comment:
We might also want this in the websocket, which requires #191. I don't think this code actually works on Whisperfish, after we started using the WS for sending messages?

Reply:
I think this is now fixed in [81caabe](https://github.com/whisperfish/libsignal-service-rs/pull/190/commits/81caabe071e11ff3c51c66bb27dbe1fee9a6b5e3)..?

(I still plan to squish the commits for cleaner commit history.)

Reply:
It's [ca14821](https://github.com/whisperfish/libsignal-service-rs/pull/190/commits/ca14821f02a7371a7bd472bcb436fe4d1a7c0ac1) after cleanup.